[GStreamer][LibWebRTC] Keep size in decoder caps synchronized with encoded image dimensions

philn opened the pull request:
#### 009d3ec372658aa49cc55aa8bac4430381d524ea
<pre>
[GStreamer][LibWebRTC] Keep size in decoder caps synchronized with encoded image dimensions
<a href="https://bugs.webkit.org/show_bug.cgi?id=277630">https://bugs.webkit.org/show_bug.cgi?id=277630</a>

Reviewed by Xabier Rodriguez-Calvar.

Update the caps of the decoder when the encoded image dimensions change. Those caps are also used by
GStreamerVideoFrameLibWebRTC and later by LibWebRTCStatsCollector. Without this change, resolution
in inbound-rtp statistics is never updated.

Based on patch from Jacek Manko &lt;jacek.manko@consult.red&gt;.
<a href="https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1380">https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1380</a>

* Source/WebCore/platform/mediastream/libwebrtc/gstreamer/GStreamerVideoDecoderFactory.cpp:
(WebCore::GStreamerWebRTCVideoDecoder::updateCapsFromImageSize):
(WebCore::GStreamerWebRTCVideoDecoder::GetCapsForFrame): Deleted.

Canonical link: <a href="https://commits.webkit.org/282115@main">https://commits.webkit.org/282115@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/25bac34beb5e5de58522c933ab1fb7cc5143c85b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/62139 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/41493 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/14731 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/66119 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/12684 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/64258 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/49179 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/13024 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/50074 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/8775 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/65208 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/38536 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/53818 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/30881 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/35206 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/11078 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/11615 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/56950 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/11383 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/67848 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/6082 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/11152 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/57451 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/6108 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/53789 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/57678 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13806 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/5015 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/37293 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/38377 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/39473 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/38122 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

calvaris commented on Source/WebCore/platform/mediastream/libwebrtc/gstreamer/GStreamerVideoDecoderFactory.cpp:
```diff
@@ -206,13 +205,22 @@ class GStreamerWebRTCVideoDecoder : public webrtc::VideoDecoder {
+            updateCapsFromImageSize(inputImage._encodedWidth, inputImage._encodedHeight);
+
+        RELEASE_ASSERT(m_caps);
+        if (UNLIKELY(!m_caps)) {
```
Isn't this redundant? Did you mean to just assert here? If you just want to assert here, you can just do `ASSERT_NOT_REACHED` before returning inside the if

philn committed:
[GStreamer][LibWebRTC] Keep size in decoder caps synchronized with encoded image dimensions
https://bugs.webkit.org/show_bug.cgi?id=277630

Reviewed by Xabier Rodriguez-Calvar.

Update the caps of the decoder when the encoded image dimensions change. Those caps are also used by
GStreamerVideoFrameLibWebRTC and later by LibWebRTCStatsCollector. Without this change, resolution
in inbound-rtp statistics is never updated.

Based on patch from Jacek Manko <jacek.manko@consult.red>.
https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1380

* Source/WebCore/platform/mediastream/libwebrtc/gstreamer/GStreamerVideoDecoderFactory.cpp:
(WebCore::GStreamerWebRTCVideoDecoder::updateCapsFromImageSize):
(WebCore::GStreamerWebRTCVideoDecoder::GetCapsForFrame): Deleted.

Canonical link: https://commits.webkit.org/282115@main
